### PR TITLE
Define a default User-Agent header

### PR DESCRIPTION
**Pull request by jabley**

### CODE REVIEW

File: lib/request-promise.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 var Q = require('q'),
   request = process.browser ? require('browser-request') : require('request'),
-  qs = require('qs');
+  qs = require('qs'),
+  userAgent = 'Performance-Platform-Client.js/' + require('../package.json')['version'],
```
Comment (easternbloc) [inline lib/request-promise.js]: why not take the name from package.json here?

so:

```
 userAgent = require('../package.json').name + require('../package.json').version,
```

??

**Comment by easternbloc** [comment]
I'll allow the array notation this time @jabley next time I'm coming for you!!!

**Comment by jabley** [comment]
Not performance-critical code

:dancers:

**Comment by easternbloc** [comment]
This is problematic so I'm going to have to revert it out.

**Pull request merged**